hera_qm/xrfi.py

```diff
@@ -1975,13 +1984,50 @@ def xrfi_run(ocalfits_file=None, acalfits_file=None, model_file=None, data_file=
                 'abscal_chi_sq_renormed_metrics2.h5': uvf_az2, 'abscal_chi_sq_flags2.h5': uvf_azf2,
                 'combined_metrics2.h5': uvf_metrics2, 'combined_flags2.h5': uvf_fws2,
                 'flags2.h5': uvf_combined2}
-    basename = qm_utils.strip_extension(os.path.basename(output_prefix))
-    for ext, uvf in uvf_dict.items():
-        if uvf is not None:
-            outfile = '.'.join([basename, ext])
-            outpath = os.path.join(dirname, outfile)
-            uvf.write(outpath, clobber=clobber)
 
+    # Determine the actual files to store
+    # We will drop kt_size / (integrations per file) files at the start and
+    # end to avoid edge effects from the convolution kernel.
+    # If this chunk includes the start or end of the night, we will write
+    # output files for those, but flag everything.
+
+    # Read metadata from first file to get integrations per file.
+    file_lists = [ocalfits_files, acalfits_files, data_files, model_files]
+    if data_files is not None:
```

Comment:
This series of conditionals seems a bit convoluted. Not sure how to make it much better, but maybe we can dispense with dtypt and just have a uvtemp = UVData/Cal and uvtemp.read() inside the first set of conditionals. I think that would be cleaner and a bit easier to follow.

Reply:
I agree.